src/production.f90

```diff
@@ -0,0 +1,306 @@
+
+            ! -----------------------------------------------------------------
+            ! Transform velocity components to semi-spectral space:
+            ! (note: vel is overwritten and invalid after this operation)
+            do nc = 1, 3
+                call fftxyp2s(vel(:, :, :, nc), svel(:, :, :, nc))
+            enddo
+
+            ! -----------------------------------------------------------------
+            ! Calculate velocity derivatives:
+
+            ! du/dx
+            call diffx(svel(:, :, :, 1), ds)
+            call fftxys2p(ds, dudx)
+
+            ! du/dy
+            call diffy(svel(:, :, :, 1), ds)
+            call fftxys2p(ds, dudy)
+
+            ! dw/dx
+            call diffx(svel(:, :, :, 3), ds)
+            call fftxys2p(ds, dwdx)
+
+            ! dv/dy
+            call diffy(svel(:, :, :, 2), ds)
+            call fftxys2p(ds, dvdy)
+
+            ! dw/dy
+            call diffy(svel(:, :, :, 3), ds)
+            call fftxys2p(ds, dwdy)
+
+        end subroutine calc_strain
+
+
+        subroutine calc_eigs(ix, iy, iz, D, V)
+            integer, intent(in)           :: ix, iy, iz
+            double precision, intent(out) :: V(3, 3), D(3)
+            double precision              :: strain(3, 3)
+
+
+            ! get local symmetrised strain matrix, i.e. 1/ 2 * (S + S^T)
+            ! where
+            !     /u_x u_y u_z\
+            ! S = |v_x v_y v_z|
+            !     \w_x w_y w_z/
+            ! with u_* = du/d* (also derivatives of v and w).
+            ! The derivatives dv/dx, du/dz, dv/dz and dw/dz are calculated
+            ! with vorticity or the assumption of incompressibility
+            ! (du/dx + dv/dy + dw/dz = 0):
+            !    dv/dx = \zeta + du/dy
+            !    du/dz = \eta + dw/dx
+            !    dv/dz = dw/dy - \xi
+            !    dw/dz = - (du/dx + dv/dy)
+            !
+            !                         /  2 * u_x  u_y + v_x u_z + w_x\
+            ! 1/2 * (S + S^T) = 1/2 * |u_y + v_x   2 * v_y  v_z + w_y|
+            !                         \u_z + w_x  v_z + w_y   2 * w_z/
+            !
+            ! S11 = du/dx
+            ! S12 = 1/2 * (du/dy + dv/dx) = 1/2 * (2 * du/dy + \zeta) = du/dy + 1/2 * \zeta
+            ! S13 = 1/2 * (du/dz + dw/dx) = 1/2 * (\eta + 2 * dw/dx) = 1/2 * \eta + dw/dx
+            ! S22 = dv/dy
+            ! S23 = 1/2 * (dv/dz + dw/dy) = 1/2 * (2 * dw/dy - \xi) = dw/dy - 1/2 * \xi
+            ! S33 = dw/dz = - (du/dx + dv/dy)
+            strain(1, 1) = dudx(iz, iy, ix)                               ! S11
+            strain(1, 2) = dudy(iz, iy, ix) + f12 * vor(iz, iy, ix, 3)    ! S12
+            strain(1, 3) = dwdx(iz, iy, ix) + f12 * vor(iz, iy, ix, 2)    ! S13
+            strain(2, 2) = dvdy(iz, iy, ix)                               ! S22
+            strain(2, 3) = dwdy(iz, iy, ix) - f12 * vor(iz, iy, ix, 1)    ! S23
+            strain(3, 3) = -(dudx(iz, iy, ix) + dvdy(iz, iy, ix))         ! S33
```

Comment:
I have checked this carefully - looks correct.  Only make sure vor is in physical space.